.github/workflows/release.yaml

```diff
@@ -124,50 +113,103 @@ jobs:
           draft: true
           prerelease: false
           fail_on_unmatched_files: true
-          files: |
-            ${{ env.ASSET }}
-            ${{ env.ASSET }}.sha256
-            ${{ env.ASSET }}.sha512
+          files: "${{ steps.build.outputs.files }}"
 
-  build-release-wasm:
-    name: build-release-wasm
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: release-${{ matrix.target }}
+          path: "${{ steps.build.outputs.files }}"
+          overwrite: true
+
+  build-container-images:
+    name: build-container-images
+    needs: [ build-release ]
     runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'push' || (github.repository == 'gleam-lang/gleam' && needs.prepare-nightly.outputs.should-run != 'false') }}
+    strategy:
+      matrix:
+        base-image:
+          - scratch
+          - erlang
+          - erlang-slim
+          - erlang-alpine
+          - elixir
+          - elixir-slim
+          - elixir-alpine
+          - node
+          - node-slim
+          - node-alpine
+
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4
+        with:
+          sparse-checkout: containers
 
-      - name: Install Rust toolchain
-        uses: actions-rs/toolchain@v1
+      - name: Download Gleam release from previous job
+        uses: actions/download-artifact@v4
         with:
-          toolchain: stable
-          target: wasm32-unknown-unknown
-          profile: minimal
-          override: true
+          pattern: release-{x86_64-unknown-linux-musl,aarch64-unknown-linux-musl}
+
+      - name: "Unpack and rearrange release"
```

Comment:
What does this do?

Reply:
Added an explanation. The approach might look a bit complicated, but makes more sense when additional files are used as with the SBoM PR.